Add --signing-key (pkcs8) arg to in-toto-run

lukpueh opened the pull request:
*EDIT 2023/12/05: rename `--pkcs8` to `--signing-key`, and test with `--use-dsse`*
Fixes part of #533 (missing: --signing-key support in in-toto-record)

----

Unlike the existing `--key`, the here added `--signing-key` supports a standard file format (PKCS8/pem), which is consistent for all supported key types and eliminates the need for an explicitly passed `--key-type`.

This PR adds a warning that `--key` is deprecated and `--signing-key` should be used instead.

In preparation for the removal of  key file loading (plus decryption via passed or prompted password) in securesystemslib,
the functionality is added to in-toto in only a few lines of code. This is possible thanks to the modern securesystemslib CryptoSigner interface.

Commit:
Add helper to load key file as sslib CryptoSigner

Upstream CryptoSigner may or may not provide API for this at some point
(see secure-systems-lab/securesystemslib#664). If not, it should at
least allow us to implement it ourselves without access to protected
names (see TODO comments).

Commit includes basic tests to load from un/encrypted pems. Test pems
come from secure-systems-lab/securesystemslib#604. Encryption pw is
hunter2.

Signed-off-by: Lukas Puehringer <lukas.puehringer@nyu.edu>

Commit:
Add --pkcs8 signing arg to in-toto-run

This is meant as replacement for --key, supporting a consistent standard
key file format (pkcs8/pem).

Signed-off-by: Lukas Puehringer <lukas.puehringer@nyu.edu>

Commit:
Simplify CryptoSigner load helper

Signed-off-by: Lukas Puehringer <lukas.puehringer@nyu.edu>

Commit:
Warn about -k, --key deprecation in in-toto-run

`warnings.warn` is for library code. Here we want to notify the cli
user directly.

Signed-off-by: Lukas Puehringer <lukas.puehringer@nyu.edu>

lukpueh commented:
Note about CI failures:
- ~with-sslib-main fail is expected, because we use an unmerged sslib feature~
- ~lint fail is unrelated and will be fixed separately~

lukpueh commented on in_toto/common_args.py:
```diff
@@ -163,6 +163,18 @@ def parse_password_and_prompt_args(args):
     ),
 }
 
+PKCS8_ARGS = ["--pkcs8"]
```
`pkcs8` is a bit technical. We could also call the arg:
- private-key
- signing-key
- key-file
- key-path
- ...


Any thoughts?

adityasaky replied:
I think I prefer `--signing-key`.

lukpueh replied:
Fair. Let's go with signing-key then.

adityasaky commented on tests/test_in_toto_run.py:
```diff
@@ -364,6 +364,41 @@ def test_main_encrypted_key_but_no_pw(self):
+    def test_pkcs8(self):
+        """Test in-toto-run, sign link with pkcs8 key file for each algo."""
+        pems_dir = Path(__file__).parent / "pems"
+        args = ["-n", "foo", "-x", "--pkcs8"]
```
should this also test with `--use-dsse` for completeness?

adityasaky reviewed (approved):
LGTM, minus minor comments.

Commit:
Rename cli arg --pkcs8 to --signing-key

As per review discussion.

Signed-off-by: Lukas Puehringer <lukas.puehringer@nyu.edu>

Commit:
Test new --signing-key arg with --use-dsse

Copy-pastes TestInTotoRunTool.test_pkcs8_signing_key to
TestInTotoRunToolWithDSSE.test_pkcs8_signing_key, adding
`--use-dsse` to the test invocation.

Signed-off-by: Lukas Puehringer <lukas.puehringer@nyu.edu>
